lib/routes/builds.js

```diff
@@ -355,18 +368,40 @@ module.exports = function (app) {
     const pkg = models.Package.fromPublish(payload);
     var version = pkg.version;
     var name = pkg.name;
+    const env = req.params.env;
+    const locale = req.params.locale || 'en-US';
+    let addingLocale = false;
 
     const versionRecord = await manager.getVersionRecord(name, version);
     if (versionRecord) {
-      // return early by sending error
-      return res.status(403).send({ message: `version record ${name}@${version} already exists for this package` });
+      // Check if the requested locale was already uploaded
+      const build = await new Promise((resolve, reject) => {
```

Comment:
you could use the `AwaitWrap` on the whole `Build` model outside the route in the scope of the outer routes function to avoid the inline promise.